CMP1903M Assessment 1 Base Code/Analyse.cs

```diff
@@ -1,54 +1,134 @@
 {
-    public static class Analyse
+    /// <summary>
+    /// Analyses text
+    /// </summary>
+    public class Analyse
     {
-        //Handles the analysis of text
+        private string _analysisText;
+        /// <summary>
+        /// Text to analyse, runs AnalyseText method when set
+        /// </summary>
+        public string AnalisisText
+        {
+            get { return _analysisText; }
+            set
+            {
+                _analysisText = value;
+                AnalyseText(_analysisText);
+            }
+        }
+        
+        /// <summary>
+        /// Keys as text elements, values as count
+        /// </summary>
+        public Dictionary<string, int> Elements { get; private set; }
+        /// <summary>
+        /// Keys as characters, values as count
+        /// </summary>
+        public Dictionary<string, int> Characters { get; private set; }
+
+        public Analyse()
+        {
+            
+        }
+        /// <summary>
+        /// Runs analysis methods
+        /// </summary>
+        /// <param name="input">Text to analyse</param>
+        protected void AnalyseText(string input)
+        {
+            Elements = CountElements(input);
+            Characters = CountChars(input);
+        }
 
-        //Method: analyseText
-        //Arguments: string
-        //Returns: list of integers
-        //Calculates and returns an analysis of the text
-        public static Dictionary<string, int> AnalyseText(string input)
+        /// <summary>
+        /// Analyses text using instances of AnalysisTool
+        /// </summary>
+        /// <param name="text">Text to analyse</param>
+        /// <returns>dict: Keys as text elements and values corresponding to count</returns>
+        protected Dictionary<string, int> CountElements(string text)
         {
             //List of integers to hold the first five measurements:
             //1. Number of sentences
             //2. Number of vowels
             //3. Number of consonants
             //4. Number of upper case letters
             //5. Number of lower case letters
-            //Initialise all the values in the list to '0'
+            
+            // Create dictionary with anonymous functions that count elements using Regex
             var tools = new Dictionary<string, Func<string, int>>()
             {
-                {"sentences", s => Regex.Matches(s, @"(\s|^)+[^.!?]*([.!?]|\S)", RegexOptions.Multiline).Count()},
+                {"sentences", s => Regex.Matches(s, @"(\s|^)+[^.!?\*]*([.!?\*]|\S)", RegexOptions.Multiline).Count()},
                 {"vowels", s => Regex.Matches(s, @"[aeiouAEIOU]", RegexOptions.Multiline).Count()},
                 {"consonants", s => Regex.Matches(s, @"[^aeiouAEIOU\W\s\d]", RegexOptions.Multiline).Count()},
                 {"upper", s => Regex.Matches(s, @"[A-Z]", RegexOptions.Multiline).Count()},
                 {"lower", s => Regex.Matches(s, @"[a-z]", RegexOptions.Multiline).Count()}

```

Comment:
You might alternatively use:
```suggestion
                {"upper", s => Regex.Matches(s, @"\p{Lu}").Count()},
                {"lower", s => Regex.Matches(s, @"\p{Ll}").Count()}
```

Reply:
Good suggestion, I didn't know this syntax. Thanks! 😄